aether-ui/aether/ui/assets/apps/pipeline/components/ContractCard.jsx

```diff
@@ -24,17 +24,25 @@ import { FormattedMessage } from 'react-intl'
 
 import ContractPublishButton from './ContractPublishButton'
 
-import { selectContract } from '../redux'
+import { selectContract, selectSection } from '../redux'
+import {
+  CONTRACT_SECTION_ENTITY_TYPES
+} from '../../utils/constants'
 
 class ContractCard extends Component {
+  onContractSelected (contract) {
+    this.props.selectContract(contract.pipeline, contract.id)
+    this.props.selectSection(CONTRACT_SECTION_ENTITY_TYPES)
+    this.props.history.push(`/${contract.pipeline}/${contract.id}/${CONTRACT_SECTION_ENTITY_TYPES}`)
+  }
```

Comment:
This is triggered by `componentDidUpdate` method in `PipelineList` component. You don't need to add this here.

Reply:
The PipelineList implementation has been removed to allow child components handle their navigation logic